### PR TITLE
Added rel=canonical <link/> tag to the <head/>

### LINKED ISSUE #605: Test Utils page is available in Google search results, but the front-end redirects the visitor
## Note from maintainers: If you'd like to work on this issue, please comment to claim it so we don't have duplication of work – thanks!

It seems like there's a problem with routing on https://reactjs.org/docs/test-utils

I'm getting redirected to Page Not Found by the router on the front-end.

`curl https://reactjs.org/docs/test-utils` returns the expected document.

It's accessible via https://reactjs.org/docs/test-utils.html as well.

### PR DESCRIPTION
**Pull request by nilfalse**
This should prevent wrong pages to appear in SERP.
Fixes #605

### CODE REVIEW

**Comment by sophiebits** [comment]
The URL https://deploy-preview-640--reactjs.netlify.com/docs/typechecking-with-proptypes in the deploy preview still shows "Page Not Found". We would want to fix that too if we are going to go with the canonical solution.

**Comment by nilfalse** [comment]
hi @sophiebits,
that was on purpose: the point is to just provide a hint to search engines while maintaining the site structure you have had so far.

**tl;dr.** IMO, a user should not face any changes on the website after we fix the issue with search engines and this is exactly what this PR does in its current state.

If you take a closer look at the page at the URL you referred to, you may notice that with this PR in place there is a `<link rel="canonical" href=... />` tag being added to the source code of the page, which is completely invisible for the end user, but plays very valuable role for most search engines.

Furthermore, I think either of the following solutions would potentially harm SEO ranks of the website:
a] introducing valid mirroring pages without `.html` postfix in their URL while also keeping those with `.html`;
b] redirecting (with code 301) users from `.html` pages to their non-`.html` counterparts.

Because:
a] would introduce repeating content (which search engines may penalize)
and b] would lower the SEO score for a page doing 301 in favour of the target of the redirect, but the target may still not be in the search engine index by the time and so not in the search results at all.

**Comment by nilfalse** [comment]
@sophiebits any ideas regarding this?

**Comment by nilfalse** [comment]
@sophiebits any news here?
